src/commands/qrcode.cpp

```diff
@@ -23,25 +23,172 @@
 #include <iostream>
 #include <bitcoin/explorer/define.hpp>
 
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
+#include "png.h"
+#include "qrencode.h"
+#endif
+
 using namespace bc::explorer;
 using namespace bc::explorer::commands;
 
+/*
+  The following constants and write_png method are based on:
+
+  https://github.com/fukuchi/libqrencode/blob/3.4/qrenc.c
+*/
+
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
+static constexpr int dpi = 72;
+static constexpr int version = 0;
+static constexpr int margin = 2;
+static constexpr QRencodeMode hint = QR_MODE_8;
+static constexpr QRecLevel level = QR_ECLEVEL_L;
+static constexpr int casesensitive = 1;
+static constexpr int INCHES_PER_METER = (100.0 / 2.54);
+static constexpr unsigned int fg_color[4] = {0, 0, 0, 255};
+static constexpr unsigned int bg_color[4] = {255, 255, 255, 255};
+
+
+static bool write_png(QRcode *qrcode, uint32_t size, std::string outfile);
```

Comment:
This should be implemented in `libbitcoin` as a utility (e.g `png.h/cpp`).

Explorer is intended as a thin shell around other libraries. We do not want to introduce material functionality in explorer that may be useful in a lower level context. As with BIP38 and BIP39 (ICU) this means that we have to drive the dependencies into the libbitcoin build as well, but that's trivial once the installer changes are generated.